storage/remote/queue_manager.go

```diff
@@ -1475,44 +1595,41 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue *queue) {
 			if !ok {
 				return
 			}
-			nPendingSamples, nPendingExemplars, nPendingHistograms := s.populateTimeSeries(batch, pendingData)
+
+			sendBatch(batch, s.qm.protoMsg, s.qm.enc, false)
+			// TODO(bwplotka): Previously the return was between popular and send, double check.
```

Comment:
I think it was right (or could be), because batch was not longer used (we copied all data to write requests protos). This could be something to improve depending on the next refactor once we merge to main, but not critical.

Nothing to do in this PR for now?